Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANS/SANSSave.py

```diff
@@ -204,12 +233,56 @@ def PyExec(self):
                 raise e
         progress.report("Finished saving workspace to files.")
 
+    def _extract_metadata(self, workspace) -> dict[str:Any]:
+        additional_properties = {}
+        sample = workspace.sample()
+        maybe_geometry = convert_to_shape(sample.getGeometryFlag())
+        additional_properties["SampleHeight"] = sample.getHeight()
+        additional_properties["SampleWidth"] = sample.getWidth()
+        additional_properties["SampleThickness"] = sample.getThickness()
+
+        instrument = SANSInstrument_string_as_key_NoInstrument[workspace.getInstrument().getName()]
+        additional_properties["DetectorNames"] = ",".join(get_detector_names_from_instrument(instrument))
+
+        if maybe_geometry is not None:
+            additional_properties["Geometry"] = maybe_geometry.value
+
+        additional_properties["BackgroundSubtractionWorkspace"] = self.getProperty("BackgroundSubtractionWorkspace").value
+        additional_properties["BackgroundSubtractionScaleFactor"] = self.getProperty("BackgroundSubtractionScaleFactor").value
+
+        return additional_properties
+
+    def _fix_polarization_props_types(self, polarization_props) -> dict[str:str]:
+        return dict([(key, polarization_props[key].value) for key in polarization_props.keys()])
+
     def validateInputs(self):
         errors = dict()
         # The workspace must be 1D or 2D if the second axis is numeric
         workspace = self.getProperty("InputWorkspace").value
-        number_of_histograms = workspace.getNumberHistograms()
-        axis1 = workspace.getAxis(1)
+        if isinstance(workspace, WorkspaceGroup):
+            return self._validate_group(errors, workspace)
+        if self.getProperty("PolarizedNXcanSAS").value:
+            errors.update({"InputWorkspace": "Must be a workspace group to save using PolarizedNXcanSAS"})
+            return errors
+        return self._validate_workspace(errors, workspace)
+
+    def _validate_group(self, errors, input_workspace):
+        for i in range(0, input_workspace.getNumberOfEntries()):
+            self._validate_workspace(errors, input_workspace.getItem(i))
+        if self.getProperty("PolarizedNXcanSAS").value:
+            self._validate_pol_props(errors)
+        if (
```

Comment:
Does it make sense to check this and the props before validating all of the workspaces?